src/Xamarin.Android.Build.Tasks/Tasks/Aot.cs

```diff
@@ -232,20 +138,13 @@ IEnumerable<Config> GetAotConfigs (NdkTools ndk)
 						Path.GetFileName (assembly.ItemSpec)));
 
 					string tempDir = Path.Combine (outdir, Path.GetFileName (assembly.ItemSpec));
-					if (!Directory.Exists (tempDir))
-						Directory.CreateDirectory (tempDir);
-
-					var aotOptions = GetAotOptions (outdir, mtriple, toolPrefix);
-					aotOptions.Add ($"outfile={outputFile}");
-					aotOptions.Add ($"llvm-path={SdkBinDirectory}");
-					aotOptions.Add ($"temp-path={tempDir}");
-
-					if (!String.IsNullOrEmpty (ldName)) {
-						// MUST be before `ld-flags`, otherwise Mono fails to parse it on Windows
-						aotOptions.Add ($"ld-name={ldName}");
-					}
+					Directory.CreateDirectory (tempDir);
 
-					aotOptions.Add ($"ld-flags={ldFlags}");
+					var aotOptions = GetAotOptions (ndk, arch, level, outdir, mtriple, toolPrefix);
+					// NOTE: ordering seems to matter on Windows
+					aotOptions.Insert (0, $"outfile={outputFile}");
+					aotOptions.Insert (0, $"llvm-path={SdkBinDirectory}");
+					aotOptions.Insert (0, $"temp-path={tempDir}");
```

Comment:
The PR build is green (just seems like Github status didn't update).

The most recent thing that broke was the ordering of these parameters. It seems like the AOT compiler wants `llvm-path` before `ld-name` or `ld-flags`.